projects/igniteui-angular/src/lib/grids/grid-navigation.service.ts

```diff
@@ -208,35 +216,37 @@ export class IgxGridNavigationService {
 
     public navigateTop(visibleColumnIndex) {
         const verticalScroll = this.grid.verticalScrollContainer.getVerticalScroll();
+        const cellSelector = this.isTreeGrid && visibleColumnIndex === 0 ? 'igx-tree-grid-cell' : 'igx-grid-cell';
         if (verticalScroll.scrollTop === 0) {
             const cells = this.grid.nativeElement.querySelectorAll(
-                `igx-grid-cell[data-visibleIndex="${visibleColumnIndex}"]`);
+                `${cellSelector}[data-visibleIndex="${visibleColumnIndex}"]`);
             cells[0].focus();
```

Comment:
Is there any chance to have no cells and cells[0] to blow up?

Reply:
This is case when form a cell click on Arrow Up + Cntr key. To this function be called we need a cell, so in case when the scrollTop is 0, there is no way to have no cells, in other case when move scroll it is possible cells to be empty(for example when have only collapsed group rows).